pkg/server/plugin/nodeattestor/azure/msi.go

```diff
@@ -0,0 +1,193 @@
+package azure
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire/pkg/common/jwtutil"
+	"github.com/spiffe/spire/pkg/common/plugin/azure"
+	spi "github.com/spiffe/spire/proto/common/plugin"
+	"github.com/spiffe/spire/proto/server/nodeattestor"
+	"github.com/zeebo/errs"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+const (
+	pluginName = "azure_msi"
+
+	tokenLeeway           = time.Minute * 5
```

Comment:
Does this have to be so long? How long are the tokens good for?

Reply:
Tokens are good for 8 hours. However, the key purpose of the leeway is so that the token, which was just barely issued (and has a not-before claim that matches the issued-at claim), can be validated in the case where the server and agent have clock differences.

Reply:
I can add a comment.